deploy/Chart/templates/rp/deployment.yaml

```diff
@@ -30,6 +30,93 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: applications-rp
+      {{- if eq .Values.global.terraform.enabled true }}
+      # Init container to pre-download Terraform binary to a shared volume
+      # This avoids downloading Terraform at runtime and improves recipe execution performance
+      initContainers:
+      - name: terraform-init
+        image: "alpine:latest"
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          echo "Terraform init container starting..."
+          echo "Running on architecture: $(uname -m)"
+          echo "Alpine version: $(cat /etc/alpine-release)"
+          
+          # Create terraform directory
+          mkdir -p "{{ .Values.rp.terraform.path }}"
+          echo "Created directory: {{ .Values.rp.terraform.path }}"
+          
+          # Detect architecture for terraform download
+          ARCH=$(uname -m)
+          case $ARCH in
+            x86_64) TERRAFORM_ARCH="amd64" ;;
+            aarch64|arm64) TERRAFORM_ARCH="arm64" ;;
+            *) echo "ERROR: Unsupported architecture: $ARCH"; exit 1 ;;
+          esac
+          echo "Terraform architecture: $TERRAFORM_ARCH"
+          
+          # Install wget and unzip if not available (Alpine doesn't include them by default)
+          if ! which wget >/dev/null 2>&1 || ! which unzip >/dev/null 2>&1; then
+            echo "Installing wget and unzip..."
+            apk add --no-cache wget unzip || { echo "ERROR: Failed to install wget/unzip"; exit 3; }
+          fi
+          
+          # Determine download URL
+          TERRAFORM_URL="{{ .Values.global.terraform.downloadUrl }}"
+          
+          # If no custom URL provided, fetch latest version from HashiCorp API
+          if [[ -z "$TERRAFORM_URL" ]]; then
+            echo "Fetching latest Terraform version from HashiCorp API..."
+            LATEST_VERSION=$(wget -qO- "https://api.releases.hashicorp.com/v1/releases/terraform/latest" | grep -o '"version":"[^"]*"' | cut -d'"' -f4)
+            if [[ -z "$LATEST_VERSION" ]]; then
+              echo "ERROR: Failed to fetch latest Terraform version from API"
+              exit 2
+            fi
+            echo "Latest Terraform version: $LATEST_VERSION"
+            TERRAFORM_URL="https://releases.hashicorp.com/terraform/${LATEST_VERSION}/terraform_${LATEST_VERSION}_linux_${TERRAFORM_ARCH}.zip"
+          fi
```

Comment:
Why do we need to do this if no custom URL is provided? **hc-install** is already going to do this. Why do we need to do both?